src/global/common.rs

```diff
@@ -478,6 +480,29 @@ pub(crate) async fn get_expose_scripts_sync_status(
     Ok((to_remove, to_add))
 }
 
+/// Check if they were all auto-exposed, or if the user manually exposed a subset of them
+pub fn check_auto_exposed(
+    env_binaries: &IndexMap<PackageName, Vec<(String, PathBuf)>>,
+    exposed_mapping_binaries: &IndexSet<Mapping>,
+) -> bool {
+    let env_binaries_names = env_binaries
+        .values()
+        .flatten()
+        .map(|(name, _)| name)
+        .collect_vec();
```

Comment:
Could you try if this also works without collecting?
My expectation would be that you can just continue with the iterator.

No big deal if it doesn't work.

Reply:
good point - changed a little and also used a better structure to verify contains :)